### ISSUE TITLE
When batch size > 1, the accuracy of MNIST training is very low?

### ISSUE THREAD

**Issue by wusaifei** [BindsNET/bindsnet#385]
Hello, I'm in conv_ mnist.py and SOM_ LM_SNN.py is accelerated, batchsize = 32, the accuracy is very low, and with the increase of training algebra, the accuracy will also decline.
![SOM_LM_SNNS](https://user-images.githubusercontent.com/27795216/85092230-db389100-b21b-11ea-8213-9824be279e9b.png)
![conv_mnist](https://user-images.githubusercontent.com/27795216/85092235-decc1800-b21b-11ea-875f-6fe247ebbe54.png)

**Comment by wusaifei** [comment]
@Hananel-Hazan @amolk learning.py , there is no kwargs ['reward '] parameter in the mstdp rule. Do you need to set it manually? Or just set it to 1

**Comment by SimonInParis** [comment]
In SOM, You can try to change the batch learning summation by providing torch.sum when building the network. When calling:
`network = IncreasingInhibitionNetwork(`
you can try adding:
`reduction = torch.sum`
(default behavior is torch.mean)
Also, make sure you are running Python3.6.x. From Python 3.7 on, it is known to be incompatible and corrupts the learning scheme.

**Comment by wusaifei** [comment]
@SimonInParis Thank you very much for your reply. There is also a small question: supervised_ mnist.py The accuracy of the test data is always very low at 10%, which is the result of random selection, but the training set is as high as 96%. What's the reason?

**Comment by Hananel-Hazan** [comment]
Thanks @SimonInParis

**Comment by wusaifei** [comment]
@Hananel-Hazan Thank you very much for your update. This is the best library for learning SNN.

**Comment by wusaifei** [comment]
@Hananel-Hazan I'm really sorry to disturb you again. The accuracy of the modified code is a little improved, but the accuracy of the test set is maintained at about 18%. Will it be because the training network has not been updated.

**Comment by Hananel-Hazan** [comment]
I pushed an update to the master branch. The parameter change will increase accuracy. But improving it further will require applying a selection mechanism instead of random selection. The selection needs to choose the best neuron from the designated class and to clamp it.

**Comment by wusaifei** [comment]
@Hananel-Hazan Thank you very much for your reply. Another problem is that in the mstdp, mstdpet and Rmax learning rules, there are reward = kwargs ["reward"], reward signal from reinforcement learning task. However, there is no "reward" parameter in * * kwargs dictionary. I know it can be float or tensor, but how should I update it? Or do I directly set a constant 1, kwargs ["reward"] = 1, reward = kwargs ["reward"]?

**Comment by wusaifei** [comment]
@Hananel-Hazan Thank you very much for your reply. Another problem is that in the mstdp, mstdpet and Rmax learning rules, there are reward = kwargs ["reward"], reward signal from reinforcement learning task. However, there is no "reward" parameter in * * kwargs dictionary. I know it can be float or tensor, but how should I update it? Or do I directly set a constant 1, kwargs ["reward"] = 1, reward = kwargs ["reward"]?

**Comment by wusaifei** [comment]
Does the author consider adding the method of FSDD speech classification? It's very different from the image.

**Comment by wusaifei** [comment]
Bindsnet is the best toolkit for learning SNN and various learning rules. Thank you very much for your open source. Does the author consider adding the code of FSDD speech classification? It's very different from the image.

**Comment by wusaifei** [comment]
Hello, author, where can I get the old version of bindsnet source code？

**Comment by Hananel-Hazan** [comment]
You need to pass the reward via the `network.run` using reward = kwargs ["reward"].
Please see the example `breakout-stdp` [link](https://github.com/BindsNET/bindsnet/blob/master/examples/breakout/breakout_stdp.py)
